### PR TITLE
feat: allow for before & after actions configuration

### PR DESCRIPTION
**Pull request by kkafar**
# Description

Right now you can specify before & after actions.

# Checklist

* [ ] Included code to test these changes
* [ ] Updated Jira

### CODE REVIEW

File: src/main/java/io/rpg/controller/PopupController.java [hunk 1]
```diff
@@ -23,28 +23,36 @@ public PopupController() {
...
+  public void openTextPopup(String text, int x, int y) {
     TextPopup popupScene = new TextPopup(text);
     popupStage.setScene(popupScene);
 
-    popupStage.show();
-
-    popupStage.setX(x - popupScene.getWidth() / 2);
-    popupStage.setY(y - popupScene.getHeight() / 2);
+    System.out.println("width: " + popupScene.getWidth());
+    System.out.println("height: " + popupScene.getHeight());
```
Comment (co012) [inline src/main/java/io/rpg/controller/PopupController.java:31]:
```suggestion
```

File: src/main/java/io/rpg/model/actions/ActionEngine.java [hunk 2]
```diff
@@ -83,7 +84,7 @@ public void acceptQuizResult(boolean correct, int pointsCount) {
     var controller = controller();
     if (correct) {
       controller.getPlayerController().addPoints(pointsCount);
-      controller.getPopupController().hidePopup();
+//      controller.getPopupController().hidePopup();
```
Comment (co012) [inline src/main/java/io/rpg/model/actions/ActionEngine.java:87]:
```suggestion
```

File: src/main/java/io/rpg/model/actions/BaseAction.java [hunk 3]
```diff
@@ -29,4 +34,26 @@ public void setEmitter(GameObject emitter) {
...
+  }
+
+  @Override
+  public void setBeforeAction(Action action) {
+    beforeAction = action;
+  }
+
+  @Override
+  @Nullable
+  public Action getBeforeAction() {
```
Comment (co012) [inline src/main/java/io/rpg/model/actions/BaseAction.java:50]: Maybe Optional<Action> would be better here?

**Review by co012 — APPROVED** [review]
Try to decrease null usage

File: src/main/java/io/rpg/controller/PopupController.java (hunk 1 above)
Comment (kkafar) [reply src/main/java/io/rpg/controller/PopupController.java]: Removed here: [c9bbf01](https://github.com/RPG-IO/rpg-io/pull/79/commits/c9bbf0187f77110a120aa3f58028e712f5c7f69c)

File: src/main/java/io/rpg/model/actions/ActionEngine.java (hunk 2 above)
Comment (kkafar) [reply src/main/java/io/rpg/model/actions/ActionEngine.java]: [c9524a6](https://github.com/RPG-IO/rpg-io/pull/79/commits/c9524a65ee629eb28f1e11cead30aac9f907932b)

File: src/main/java/io/rpg/model/actions/BaseAction.java (hunk 3 above)
Comment (kkafar) [reply src/main/java/io/rpg/model/actions/BaseAction.java]: Good idea. Done here: [38c65fc](https://github.com/RPG-IO/rpg-io/pull/79/commits/38c65fce529153f5958672e365652700927e97ea)

**Pull request merged**